pkg/apis/apidiscovery/types.go

```diff
@@ -0,0 +1,146 @@
+	Scope ResourceScope
+	// singularResource is the singular name of the resource.  This allows clients to handle plural and singular opaquely.
+	// For many clients the singular form of the resource will be more understandable to users reading messages and should be used when integrating the name of the resource into a sentence.
+	// The command line tool kubectl, for example, allows use of the singular resource name in place of plurals.
+	// The singular form of a resource should always be an optional element - when in doubt use the canonical resource name.
+	SingularResource string
+	// verbs is a list of supported API operation types (this includes
+	// but is not limited to get, list, watch, create update, patch,
+	// delete, deletecollection, and proxy)
+	// +listType=set
+	Verbs []string
+	// shortNames is a list of suggested short names of the resource.
+	// +listType=set
+	ShortNames []string
+	// categories is a list of the grouped resources this resource belongs to (e.g. 'all').
+	// Clients may use this to simplify acting on multiple resource types at once.
+	// +listType=set
+	Categories []string
+	// subresources is a list of subresources provided by this resource. Subresources are located at /apis/<APIGroupDiscovery.objectMeta.name>/<APIVersionDiscovery.version>/<APIResourceDiscovery.Resource>/name-of-instance/<APIResourceDiscovery.subresources[i].subresource>
+	// +listType=map
+	// +listMapKey=subresource
+	Subresources []APISubresourceDiscovery
+}
+
+// ResourceScope is an enum defining the different scopes available to a resource.
+type ResourceScope string
+
+const (
+	ScopeCluster   ResourceScope = "Cluster"
+	ScopeNamespace ResourceScope = "Namespaced"
+)
+
+// DiscoveryFreshness is an enum defining whether the Discovery document published by an apiservice is up to date (fresh).
+type DiscoveryFreshness string
+
+const (
+	DiscoveryFreshnessCurrent DiscoveryFreshness = "Current"
+	DiscoveryFreshnessStale   DiscoveryFreshness = "Stale"
+)
+
+// APISubresourceDiscovery provides information about an API subresource for discovery.
+type APISubresourceDiscovery struct {
+	// subresource is the name of the subresource.  This is used in the URL path and is the unique identifier
+	// for this resource across all versions.
+	Subresource string
+	// responseKind describes the type of serialization that will be returned from this endpoint.
+	// Some subresources do not return normal resources, these will have nil return types.
+	ResponseKind *v1.GroupVersionKind
+	// acceptedTypes describes the kinds that this endpoint accepts.  It is possible for a subresource to accept multiple kinds.
+	// It is also possible for an endpoint to accept no standard types.  Those will have a zero length list.
+	// +listType=map
+	// +listMapKey=group
+	// +listMapKey=version
+	// +listMapKey=kind
+	AcceptedTypes []v1.GroupVersionKind
+	// verbs is a list of supported kube verbs: get, list, watch, create,
+	// update, patch, delete
```

Comment:
add `deletecollection`, also make it clear this is just an example and other verbs can be included (since we're aggregating from custom backends)

Reply:
Added